pkg/cmd/pulumi/destroy.go

```diff
@@ -143,9 +143,13 @@ func newDestroyCmd() *cobra.Command {
 				return result.FromError(fmt.Errorf("getting stack configuration: %w", err))
 			}
 
+			snap, err := s.Snapshot(commandContext())
```

Comment:
Is there any performance concern here? Is calling Snapshot always fast? Does this cause us to fully read here and again when the engine starts? Does that matter?

Reply:
I don't know if calling snapshot is fast, but it is cached. This change shouldn't incur a general performance cost. 
https://github.com/pulumi/pulumi/blob/eb975813f44c42cb6ea0eb02af33c339e181a95e/pkg/backend/httpstate/stack.go#L121-L133
https://github.com/pulumi/pulumi/blob/eb975813f44c42cb6ea0eb02af33c339e181a95e/pkg/backend/filestate/stack.go#L55